lib/grandpa/grandpa.go

```diff
@@ -1155,3 +1155,37 @@ func (s *Service) findParentWithNumber(v *Vote, n uint32) (*Vote, error) {
 
 	return NewVoteFromHeader(b), nil
 }
+
+// GetSetID returns the current setID
+func (s *Service) GetSetID() uint64 {
+	return s.state.setID
+}
+
+// GetRound return the current round number
+func (s *Service) GetRound() uint64 {
+	s.roundLock.Lock()
+	defer s.roundLock.Unlock()
+
+	return s.state.round
+}
+
+// GetVoters returns the list of current grandpa.Voters
+func (s *Service) GetVoters() Voters {
+	return s.state.voters
+}
+
+// PreVotes returns the current prevotes to the current round
+func (s *Service) PreVotes() map[ed25519.PublicKeyBytes]*Vote {
+	s.mapLock.Lock()
+	defer s.mapLock.Unlock()
+
+	return s.prevotes
+}
```

Comment:
this should also return the votes in `pvEquivocations`

Reply:
Done!